tests/lib/rules/no-unused-vars.js

```diff
@@ -179,6 +179,18 @@ ruleTester.run("no-unused-vars", rule, {
             options: [{ vars: "all", args: "all" }]
         },
 
+        // Using object rest for variable omission
+        {
+            code: "const data = { type: 'coords', x: 1, y: 2 };\nconst { type, ...coords } = data;\n console.log(coords);",
+            options: [{ ignoreRestSiblings: true }],
+            parserOptions: {
+                ecmaVersion: 8,
+                ecmaFeatures: {
+                    experimentalObjectRestSpread: true
+                }
+            }
+        },
```

Comment:
Can you add a test that has code like this, but without the `ignoreRestSiblings` option enabled? It would be useful to verify that an error is reported in that case.